File: VGTU-Aventory-Frontend/src/features/dashboard/types/product.ts
export type ApiProduct = {
  productId: number;
  productName: string;
  owner: string | null;
  sales: unknown[];
  price: number;
  productDescription: string;
  photoUrl: string;
  quantity: number;
  qrCode: string;
  category?: {
    categoryId?: number;
    categoryName?: string;
  };
  categoryId?: number;
  categoryName?: string;
};

export type Product = {
  id: string;
  name: string;
  quantity: number;
  description: string;
  photoUrl: string;
  qrCode: string;
  price: number;
  categoryId?: number;
  categoryName?: string;
};

export function mapApiProduct(p: ApiProduct): Product {
  const categoryId = p.categoryId ?? p.category?.categoryId;
  const categoryName = p.categoryName ?? p.category?.categoryName;

  return {
    id: String(p.productId),
    name: p.productName,
    quantity: p.quantity,
    description: p.productDescription,
    photoUrl: p.photoUrl,
    qrCode: p.qrCode,
    price: p.price,
    categoryId,
    categoryName,
  };
}

File: VGTU-Aventory-Backend/src/main/java/org/example/vgtuaventory/model/Product.java
package org.example.vgtuaventory.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int productId;

    @Column(unique = true)
    private String productName;

    @ManyToOne
    private User owner;

    @OneToMany(mappedBy = "product", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    @JsonIgnore
    protected User owner;
    @OneToMany(mappedBy ="product", cascade = CascadeType.ALL, fetch = FetchType.LAZY)
    private List<Sale> sales;
    private Double price;
    private String productDescription;
    private String photoUrl;
    private int quantity;
    private String qrCode;
    protected Double price;
    protected String productDescription;
    protected String photoUrl;
    protected int quantity;
    protected String qrCode; //veliau zaisim
    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;
}

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;
}